internal/controllers/topology/cluster/reconcile_state_test.go

```diff
@@ -632,10 +726,31 @@ func TestReconcileControlPlaneObject(t *testing.T) {
 			g.Expect(err).ToNot(HaveOccurred())
 
 			// Create ControlPlane object for fetching data into
-			gotControlPlaneObject := builder.ControlPlane("", "").Build()
+			gotControlPlaneObject := builder.TestControlPlane("", "").Build()
 			err = env.GetAPIReader().Get(ctx, client.ObjectKeyFromObject(tt.want.Object), gotControlPlaneObject)
 			g.Expect(err).ToNot(HaveOccurred())
 
+			// check for template rotation.
+			gotRotation := false
+			var gotInfrastructureMachineRef *corev1.ObjectReference
+			if tt.class.InfrastructureMachineTemplate != nil {
+				gotInfrastructureMachineRef, err = contract.ControlPlane().MachineTemplate().InfrastructureRef().Get(gotControlPlaneObject)
+				g.Expect(err).ToNot(HaveOccurred())
+				if tt.original != nil {
+					if tt.original.InfrastructureMachineTemplate != nil && tt.original.InfrastructureMachineTemplate.GetName() != gotInfrastructureMachineRef.Name {
+						gotRotation = true
+					}
+				}
+			}
+			g.Expect(gotRotation).To(Equal(tt.wantRotation))
+
+			// if template has been rotated, fixup infrastructureRef in the wantControlPlaneObjectSpec before comparison.
+			if gotRotation {
+				gotInfrastructureMachineRef, err := contract.ControlPlane().MachineTemplate().InfrastructureRef().Get(gotControlPlaneObject)
+				g.Expect(err).ToNot(HaveOccurred())
```

Comment:
```suggestion
```
nit: I think this is just the same code as in l.737/l.738, so we don't need it here again
We could also just move l.751 after l.741

Reply:
done